src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+            Span<byte> uidSak = stackalloc byte[7];
+            Span<byte> uid = stackalloc byte[10];
+            Span<byte> sak = stackalloc byte[1];
+            Span<byte> sakInterm = stackalloc byte[5];
+            int numBytes;
+
+            DateTime dtTimeout = DateTime.Now.AddMilliseconds(timeoutPollingMilliseconds);
+
+            try
+            {
+                // Switches off the CRC off in RX and TX direction
+                CrcReceptionTransfer = false;
+                do
+                {
+                    // Clears all interrupt
+                    SpiWriteRegister(Command.WRITE_REGISTER, Register.IRQ_CLEAR, new byte[] { 0xFF, 0xFF, 0x0F, 0x00 });
+                    // Sets the PN5180 into IDLE state
+                    SpiWriteRegister(Command.WRITE_REGISTER_AND_MASK, Register.SYSTEM_CONFIG, new byte[] { 0xF8, 0xFF, 0xFF, 0xFF });
+                    // Activates TRANSCEIVE routine
+                    SpiWriteRegister(Command.WRITE_REGISTER_OR_MASK, Register.SYSTEM_CONFIG, new byte[] { 0x03, 0x00, 0x00, 0x00 });
+                    // Sends REQB command
+                    ret &= SendDataToCard(new byte[] { 0x26 }, 7);
+                    (numBytes, _) = GetNumberOfBytesReceivedAndValidBits();
+                    if (numBytes > 0)
+                    {
+                        ret &= ReadDataFromCard(atqa, atqa.Length);
+                        if (!ret)
+                        {
+                            return false;
+                        }
+                    }
+                    else if (dtTimeout < DateTime.Now)
+                    {
+                        return false;
+                    }
+                }
+                while (numBytes == 0);
+
+                card.Atqa = BinaryPrimitives.ReadUInt16LittleEndian(atqa);
+                int numberOfUid = 0;
+                // We do have a card! Now send anticollision
+                // There are 3 SL maximum. For looping and adjusting the SL
+                // SL1 = 0x93, SL2 = 0x95, SL3 = 0x97
+                for (int i = 0; i < 3; i++)
+                {
+                    // Select SL1
+                    uidSak[0] = (byte)(0x93 + i * 2);
+                    // NVB = Number of valid bits
+                    uidSak[1] = 0x20;
+                    SendDataToCard(uidSak.Slice(0, 2));
+                    // Read 5 bytes sak. Byte 1 will tell us if we have the full UID or if we need to read more
+                    ReadDataFromCard(sakInterm.Slice(0, 5));
+                    // Switches back on the CRC off in RX and TX direction
+                    CrcReceptionTransfer = true;
+                    // Now finish the anticollision with the new NVB
+                    uidSak[1] = 0x70;
+                    // Add the previous elements to this last anticollision to send
+                    sakInterm.Slice(0, 5).CopyTo(uidSak.Slice(2));
+                    ret &= SendDataToCard(uidSak);
+
```

Comment:
you can probably `if (!ret) return false` here

Reply:
Yes and no, I've seen weird behaviors sometimes which I think I fixed by adjusting correctly the various timings. You may miss writing but still receive something from the previous time. So I4ll just keep it like this and I changed &= for a simple = which makes more sense.

Reply:
fair enough